src/lib/radio/radio.ts

```diff
@@ -124,14 +126,31 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
   }
 
   /** Whether the labels should appear after or before the radio-buttons. Defaults to 'after' */
-  @Input() labelPosition: 'before' | 'after' = 'after';
+  _labelPosition: 'before' | 'after' = 'after';
+  @Input()
+  get labelPosition(): 'before' | 'after' {
+    return this._labelPosition;
+  }
+  set labelPosition(v: 'before' | 'after') {
+    this._labelPosition = v
+    if (this._radios) {
```

Comment:
This shouldn't be necessary; the individual radios have a getter that reads the value from the group if it exists without clobbering its own value.